src/lib/utils/should-publish.ts

```diff
@@ -0,0 +1,17 @@
+import Entry from '../entities/entry'
+
+export default function shouldPublish (shouldPublish: boolean | 'preserve', entry: Entry) {
+  if (shouldPublish === true) {
+    return true
+  } else if (shouldPublish === false) {
+    return false
+  } else if (shouldPublish === 'preserve') {
+    if (!entry.publishedVersion || !entry.version || isNaN(entry.version)) {
+      return false
+    }
+    const hasUnpublishedChanges = !!entry.publishedVersion && entry.version > entry.publishedVersion + 2
```

Comment:
let's just add some sort of note or documentation about this somewhere. then looks good to merge